src/Upload.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { DetailedHTMLProps, InputHTMLAttributes, ReactNode, useId } from "react";
+import { createComponentI18nApi } from "./i18n";
+import { symToStr } from "tsafe/symToStr";
+import { cx } from "./tools/cx";
+import { fr } from "./fr";
+import { Equals, assert } from "tsafe";
+
+export type UploadProps = {
+    classes?: any;
+    className?: string;
+    /** @default false */
+    disabled?: boolean;
+    hint?: string;
+    /** @default false */
+    multiple?: boolean;
+    /** @default "default" */
+    state?: "success" | "error" | "default";
+    /** The message won't be displayed if state is "default" */
+    stateRelatedMessage?: ReactNode;
+    /** Props forwarded to the underlying <input /> element */
+    nativeInputProps?: DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>;
+};
+
+const Upload = (props: UploadProps) => {
```

Comment:
I'll understand if you decide not to memoize this component. 
Even though our tendency is to memoize every component, in this case, it's not be beneficial. 
However, enabling forwarding a ref is something we must provide for every component.